Frontend/ui-driver/src/Components/DueDetailsList/View.purs

```diff
@@ -0,0 +1,198 @@
+
+  as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+  is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+  or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+  the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Components.DueDetailsList.View where
+
+import Common.Types.App
+import Components.DueDetailsList.Controller
+
+import Data.Array (mapWithIndex)
+import Effect (Effect)
+import Font.Size as FontSize
+import Font.Style as FontStyle
+import Language.Types (STR(..))
+import Prelude (Unit, (==), const, (<>), (&&), bind, ($), pure, unit, (/=), void)
+import PrestoDOM (Gravity(..), Length(..), Margin(..), Orientation(..), Visibility(..), PrestoDOM, Padding(..), alignParentBottom, padding, background, color, cornerRadius, fontStyle, gravity, height, imageUrl, imageView, linearLayout, margin, onClick, orientation, stroke, text, textSize, textView, weight, width, imageWithFallback, lottieAnimationView, id, afterRender, visibility)
+import Styles.Colors as Color
+
+view :: forall w . (Action -> Effect Unit) -> DueDetailsListState -> PrestoDOM (Effect Unit) w
+view push state =
+    linearLayout
+    [ width MATCH_PARENT
+    , height MATCH_PARENT
+    ][ linearLayout
+       [ width MATCH_PARENT
+       , height MATCH_PARENT
+       , orientation VERTICAL
+       , margin (Margin 16 10 16 10)
+       ](mapWithIndex
+         (\index item ->
+          linearLayout
+          [ width MATCH_PARENT
+          , height WRAP_CONTENT
+          , weight 1.0
+          , orientation VERTICAL
+          , padding $ Padding 16 20 16 20
+          , gravity CENTER
+          , stroke if state.activeIndex == index then ("1,"<>Color.blue800) else ("1,"<>Color.grey900)
+          , onClick push (const (SelectDue index))
+          , margin $ MarginBottom 16
+          , cornerRadius 10.0
+          , background if state.activeIndex == index then Color.blue600 else Color.white900
+          ][ 
+            linearLayout[
+              width MATCH_PARENT
+              , height WRAP_CONTENT
+              , margin $ MarginBottom if state.activeIndex == index then 16 else 4
+            ]
+            [
+              linearLayout[
+                orientation VERTICAL
+                , weight 1.0
+              ][
+                linearLayout[][
+                  textView [
+                    text "Trip Date " 
+                    , textSize FontSize.a_12
+                    , fontStyle $ FontStyle.medium LanguageStyle
+                    , color Color.black700
+                  ]
+                  ,textView [
```

Comment:
As the textView is used at multiple places 
You can create a single textView function and use that everywhere